src/components/KeyboardShortcuts/index.spec.tsx

```diff
@@ -0,0 +1,181 @@
+import * as React from 'react';
+import { Store } from 'redux';
+
+import configureStore from '../../configureStore';
+import { RelativePathPosition } from '../../reducers/fileTree';
+import { actions as versionActions } from '../../reducers/versions';
+import {
+  createFakeKeyboardEvent,
+  createFakeThunk,
+  shallowUntilTarget,
+  spyOn,
+} from '../../test-helpers';
+import styles from './styles.module.scss';
+
+import KeyboardShortcuts, {
+  DefaultProps,
+  PublicProps,
+  KeyboardShortcutsBase,
+} from '.';
+
+describe(__filename, () => {
+  type RenderParams = {
+    store?: Store;
+  } & Partial<PublicProps & DefaultProps>;
+
+  const render = ({
+    currentPath = 'file1.js',
+    pathList = ['file1.js'],
+    store = configureStore(),
+    versionId = 123,
+    ...moreProps
+  }: RenderParams = {}) => {
+    const props = {
+      _goToRelativeFile: jest.fn(),
+      currentPath,
+      pathList,
+      versionId,
+      ...moreProps,
+    };
+
+    return shallowUntilTarget(
+      <KeyboardShortcuts {...props} />,
+      KeyboardShortcutsBase,
+      {
+        shallowOptions: { context: { store } },
+      },
+    );
+  };
+
+  it('renders a list of keyboard shortcuts', () => {
+    const root = render();
+
+    expect(root.find(`.${styles.KeyboardShortcuts}`)).toHaveLength(1);
+  });
+
+  it('binds and unbinds keydown to the listener', () => {
+    document.addEventListener = jest.fn();
```

Comment:
This looks like it could pollute the context of other tests if they also access `document`. Are you relying on some magic jest behavior that gives us `document` in a sandbox or something like that?

I'd rather see this pass in a fake document like `render({ _document })`

Reply:
Good idea, and no, I wasn't relying on any jest magic.